src/components/taxonomicService/taxonomicServiceFormComponents/FormBuilder.tsx

```diff
@@ -0,0 +1,207 @@
+/* Import Dependencies */
+import { Formik, Form } from "formik";
+import jp from 'jsonpath';
+import { Row, Col } from 'react-bootstrap';
+
+/* Import Types */
+import { FormField, Dict } from "app/Types";
+
+/* Import Components */
+import BooleanField from "./BooleanField";
+import DateField from "./DateField";
+import FormBuilderFieldArray from "./FormBuilderFieldArray";
+import MultiSelectField from "./MultiSelectField";
+import SelectField from "./SelectField";
+import StringField from "./StringField";
+import StringArrayField from "./StringArrayField";
+import TextField from "./TextField";
+import { Button } from "components/general/CustomComponents";
+
+
+/* Props Type */
+type Props = {
+    formTemplate: {
+        [formSection: string]: {
+            title: string,
+            type: string,
+            jsonPath?: string,
+            fields: FormField[]
+        }
+    }
+};
+
+
+/**
+ * Component that renders a form builder for rendering forms based on JSON files
+ * @param formTemplate The form template to build the form from
+ * @returns JSX Component
+ */
+const FormBuilder = (props: Props) => {
+    const { formTemplate } = props;
+
+    /* Base variables */
+    const formSections: {
+        [section: string]: {
+            type: string,
+            jsonPath: string,
+            fields: FormField[]
+        }
+    } = {};
+    const initialFormValues: Dict = {};
+
+    /**
+     * Function to flatten a JSON path
+     * @returns flattened JSON path string
+     */
+    const FlattenJSONPath = (jsonPath: string): string => {
+        return jsonPath.replaceAll('[', '_').replaceAll(']', '').replaceAll("'", '');
+    };
+
+    /**
+     * Function to determine the initial form field type 
+     * @param fieldType
+     */
+    const DetermineInitialFormValue = (fieldType: string) => {
+        switch (fieldType) {
+            case 'boolean':
+                return false;
+            case 'array':
+                return [];
+            case 'multi-string':
+                return [''];
+            case 'ror':
+                return {
+                    "schema:identifier": '',
+                    "schema:name": ''
+                };
+            default:
+                return '';
+        };
+    };
+
+    /* Construct initial form values */
+    Object.entries(formTemplate).forEach(([_key, formSection]) => {
+        formSections[formSection.title] = {
+            type: formSection.type,
+            jsonPath: formSection.jsonPath ?? '',
+            fields: []
+        };
+
+        /* If is array, push to initial form values */
+        if (formSection.type === 'array') {
+            jp.value(initialFormValues, formSection.jsonPath ?? '', []);
```

Comment:
I think the following might be better at handling null values? (might be wrong though)

`jp.value(values, field.jsonPath) ?? '';`